### PR TITLE
Pad: don't error when unused fill value is zero

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #73434
* __->__ #76307

Fixes pytorch/vision#5873

In the python version of `F.pad`, checking that the fill value was
left as default was done by comparing against zero:
https://github.com/pytorch/pytorch/blob/bc2c6edaf163b1a1330e37a6e34caf8c553e4755/torch/nn/functional.py#L4366

So if someone does explicitly pass in a zero-value, then this
`TORCH_CHECK` was an accidental BC-break.

### CODE REVIEW

**Review by datumbox — COMMENTED** [review]
Thanks for the change @peterbell10. I confirm that this should resolve the BC problem and fix the downstream issues at TorchVision. 


The only question I have is on whether we should be raising a deprecation warning. Hence, I'll leave it to @albanD to approve and merge if the current approach aligns with Core's deprecation policy.

File: aten/src/ATen/native/PadNd.cpp
Comment (albanD) [inline aten/src/ATen/native/PadNd.cpp:171]: Do we really want to deprecate this? Or are we just happy with allowing `0.` here?
Given that people use this already and this is not very harmful. I would vote for not changing the current behavior. At least for now so that we can quickly land this. We can discuss (with Joel) if we want to deprecate that later.

**Review by jbschlosser — COMMENTED** [review]
Thanks for the quick forward fix!

re: Vasilis' question: Our policy in core is to throw deprecation warnings using [TORCH_WARN_ONCE](https://github.com/pytorch/pytorch/wiki/Developer-FAQ#when-should-i-torch_warn-vs-torch_warn_once) that are actionable to silence the warning. The warning here can be silenced by passing `None`  instead of `0` to torch's `pad`.

That said, it seems like torchvision's public API for `pad` would have to change to pass `None` as the default value to torch's pad. Otherwise, this warning will be thrown under normal usage with no way to remove it - is this true?

If so, maybe we should just avoid the deprecation for now to minimize pain. As @albanD mentioned, it isn't a particularly harmful thing to pass an unused fill value.

File: aten/src/ATen/native/PadNd.cpp
Comment (datumbox) [reply aten/src/ATen/native/PadNd.cpp:171]: I agree that maintaining the current behaviour until we figure out what to do is probably preferable. 

Note that we already have a PR at https://github.com/pytorch/vision/pull/5875 which mitigates the issue by not passing any parameter in specific modes but maintaining BC is worth it for other users.

Comment (peterbell10) [reply aten/src/ATen/native/PadNd.cpp]: Okay, I've removed the warning. I would argue that a passing parameter that gets ignored is somewhat confusing, but if it's already baked into torchvision then I guess it's too late to fix.

**Review by albanD — APPROVED** [review]
SGTM!

**Review by jbschlosser — APPROVED** [review]

**Review by datumbox — APPROVED** [review]
LGTM, thanks!

FYI TorchVision is going to switch to passing the value only when needed (see https://github.com/pytorch/vision/pull/5875), so that we can fix ASAP our CI.

**Pull request closed without merging**
